app/design/frontend/rwd/default/template/configurableswatches/catalog/media/js.phtml

```diff
@@ -18,11 +18,20 @@
 ?>
 
 <script type="text/javascript">
-    $j(document).on('product-media-loaded', function() {
+    if(!(typeof $j == 'function' && $j === jQuery)) {
         ConfigurableMediaImages.init('<?php echo $this->getImageType(); ?>');
         <?php foreach ($this->getProductImageFallbacks() as $imageFallback): ?>
-        ConfigurableMediaImages.setImageFallback(<?php echo $imageFallback['product']->getId(); ?>, $j.parseJSON('<?php echo $imageFallback['image_fallback']; ?>'));
+        ConfigurableMediaImages.setImageFallback(<?php echo $imageFallback['product']->getId(); ?>, JSON.parse('<?php echo $imageFallback['image_fallback']; ?>'));
         <?php endforeach ?>
-        $j(document).trigger('configurable-media-images-init', ConfigurableMediaImages);
-    });
+        document.dispatchEvent(new Event('configurable-media-images-init'));
+    } else {
+        // compatibility with rwd theme ProductMediaManager in app.js
+        $j(document).on('product-media-loaded', function() {
+            ConfigurableMediaImages.init('<?php echo $this->getImageType(); ?>');
+            <?php foreach ($this->getProductImageFallbacks() as $imageFallback): ?>
+            ConfigurableMediaImages.setImageFallback(<?php echo $imageFallback['product']->getId(); ?>, $j.parseJSON('<?php echo $imageFallback['image_fallback']; ?>'));
+            <?php endforeach ?>
+            $j(document).trigger('configurable-media-images-init', ConfigurableMediaImages);
+        });
```

Comment:
do you think it could be possible to rewrite part of the app.js in order not to have this "compatibility code" here and not to check if jquery is loaded etc?

Reply:
I had looked at it and I've also tried to bring everything configurable swatches to the base theme but it became very complicated and difficult